common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java

```diff
@@ -800,6 +803,74 @@ public void putValueInSuccessEvent(final @NonNull ApiEndEvent event, final @NonN
                 });
     }
 
+
+    /**
+     * Checks if QR + PIN authorization is available.
+     *
+     * @return true if if QR + PIN authorization is available. False otherwise.
+     */
+    @Override
+    public boolean isQrPinAvailable() throws BaseException {
+        final String methodTag = TAG + ":isQrPinAvailable";
+        return mBrokerOperationExecutor.execute(
+                null,
+                new BrokerOperation<Boolean>() {
+
+                    @Override
+                    public void performPrerequisites(final @NonNull IIpcStrategy strategy) throws BrokerCommunicationException {
+                        // AccountManagerAddAccountStrategy and BoundServiceStrategy are not supported for this operation.
+                        // We are failing fast here to avoid unnecessary IPC calls.
+                        // IPC calls through BoundServiceStrategy takes approximate 25s to timeout.
+                        if (strategy instanceof BoundServiceStrategy || strategy instanceof AccountManagerAddAccountStrategy) {
+                            final String errorMessage = strategy + " is not supported for isQrPinAvailable operation";
+                            Logger.warn(methodTag,  errorMessage);
+                            throw new BrokerCommunicationException(
+                                    BrokerCommunicationException.Category.OPERATION_NOT_SUPPORTED_ON_CLIENT_SIDE,
```

Comment:
OPERATION_NOT_SUPPORTED_ON_SERVER_SIDE?

Reply:
we could call directly the content provider strategy and ignore the rest, but I prefer to leave it this way, in case we adopt a new strategy on the future, also in this way we can have all the span data from the BrokerOperationExecutor,